src/libraries/System.Runtime.Extensions/tests/System/IO/Path.RemoveRedundantSegments.cs

```diff
@@ -0,0 +1,503 @@
+// Licensed to the .NET Foundation under one or more agreements.
```

Comment:
@jkotas another question - Is `System.Runtime.Extensions` the correct location from my unit tests, or should I also move them to `System.Runtime`? It's not entirely clear to me why the refs were moved, so I'm not sure if I should move anything else from this PR.

Reply:
Keep them under System.Runtime.Extensions. We are keeping the tests in their historic contract.

Reply:
I ask mainly because after I moved my refs to `System.Runtime.cs`, I am unable to run my unit tests located in `System.Runtime.Extensions.csproj`.

I get the errors by executing either one of these commands:
```
PS D:\runtime> dotnet msbuild .\src\libraries\System.Runtime.Extensions\tests\

PS D:\runtime> dotnet msbuild /t:RebuildAndTest .\src\libraries\System.Runtime.Extensions\tests\System.Runtime.Extensions.Tests.csproj

```

This is the output:
```
Microsoft (R) Build Engine version 16.3.0+0f4c62fea for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

C:\Users\calope\.nuget\packages\microsoft.net.compilers.toolset\3.5.0-beta3-20114-02\tasks\netcoreapp3.1\Microsoft.CSharp.Core.targets(59,5): error MSB4062: The "Microsoft.CodeAnalysis.BuildTasks.Csc" task could not be loaded from the assembly C:\Users\calope\.nuget\packages\microsoft.net.compilers.toolset\3.5.0-beta3-20114-02\build\..\tasks\netcoreapp3.1\Microsoft.Build.Tasks.CodeAnalysis.dll. Could not load file or assembly 'System.Runtime, Version=4.2.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [D:\runtime\src\libraries\System.Runtime.Extensions\tests\TargetFrameworkNameTestApp\TargetFrameworkNameTestApp.csproj]
C:\Users\calope\.nuget\packages\microsoft.net.compilers.toolset\3.5.0-beta3-20114-02\tasks\netcoreapp3.1\Microsoft.CSharp.Core.targets(59,5): error MSB4062: The "Microsoft.CodeAnalysis.BuildTasks.Csc" task could not be loaded from the assembly C:\Users\calope\.nuget\packages\microsoft.net.compilers.toolset\3.5.0-beta3-20114-02\build\..\tasks\netcoreapp3.1\Microsoft.Build.Tasks.CodeAnalysis.dll. Could not load file or assembly 'System.Runtime, Version=4.2.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [D:\runtime\src\libraries\System.Runtime.Extensions\tests\VoidMainWithExitCodeApp\VoidMainWithExitCodeApp.csproj]
C:\Users\calope\.nuget\packages\microsoft.net.compilers.toolset\3.5.0-beta3-20114-02\tasks\netcoreapp3.1\Microsoft.CSharp.Core.targets(59,5): error MSB4062: The "Microsoft.CodeAnalysis.BuildTasks.Csc" task could not be loaded from the assembly C:\Users\calope\.nuget\packages\microsoft.net.compilers.toolset\3.5.0-beta3-20114-02\build\..\tasks\netcoreapp3.1\Microsoft.Build.Tasks.CodeAnalysis.dll. Could not load file or assembly 'System.Runtime, Version=4.2.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [D:\runtime\src\libraries\System.Runtime.Extensions\tests\TestApp\TestApp.csproj]
C:\Users\calope\.nuget\packages\microsoft.net.compilers.toolset\3.5.0-beta3-20114-02\tasks\netcoreapp3.1\Microsoft.CSharp.Core.targets(59,5): error MSB4062: The "Microsoft.CodeAnalysis.BuildTasks.Csc" task could not be loaded from the assembly C:\Users\calope\.nuget\packages\microsoft.net.compilers.toolset\3.5.0-beta3-20114-02\build\..\tasks\netcoreapp3.1\Microsoft.Build.Tasks.CodeAnalysis.dll. Could not load file or assembly 'System.Runtime, Version=4.2.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [D:\runtime\src\libraries\Common\tests\CoreFx.Private.TestUtilities\CoreFx.Private.TestUtilities.csproj]
C:\Users\calope\.nuget\packages\microsoft.net.compilers.toolset\3.5.0-beta3-20114-02\tasks\netcoreapp3.1\Microsoft.CSharp.Core.targets(59,5): error MSB4062: The "Microsoft.CodeAnalysis.BuildTasks.Csc" task could not be loaded from the assembly C:\Users\calope\.nuget\packages\microsoft.net.compilers.toolset\3.5.0-beta3-20114-02\build\..\tasks\netcoreapp3.1\Microsoft.Build.Tasks.CodeAnalysis.dll. Could not load file or assembly 'System.Runtime, Version=4.2.2.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [D:\runtime\src\libraries\System.Runtime.Extensions\tests\AssemblyResolveTestApp\AssemblyResolveTestApp.csproj]
```

Reply:
You may need to do a clean build to fix this.

Reply:
I ran:
`git clean -fdx`
`coreclr -c Release`
`libraries /p:CoreCLRConfiguration=Release`
I also ran separately:
`libraries -buildtests /p:CoreCLRConfiguration=Release` but I still can't run the individual test project.

Reply:
Run `build.cmd -c Release` first. That is guaranteed to work because of it is what the CI does.